lib/src/lifecycle_module.dart

```diff
@@ -683,7 +710,10 @@ abstract class LifecycleModule extends SimpleModule with Disposable {
       if (isUnloading) {
         unloadingTransitionFuture = _transition.future;
       } else {
-        var pendingTransition = _transition?.future;
+        Future pendingTransition;
+        if (_transition != null && !_transition.isCompleted) {
+          pendingTransition = _transition?.future;
```

Comment:
I'm wondering now if this is really necessary? If it's null, the handlers that these "pending transitions" get passed to already null-check, and if it's already completed, it's just effectively an `await new Future(() {});`

Reply:
Ignore this, Todd pointed out that keeping a pending transition around indefinitely will just pass the error on whenever it is checked